plugins/bgp/bgp-plugin.go

```diff
@@ -40,12 +63,135 @@ func (p *BgpPlugin) Init() error {
+// but on the github for cn infra, there is one
+func (p *BgpPlugin) onChange(resp datasync.ProtoWatchResp) {
+	//key := resp.GetKey()
+
+	//Getting ip
+	value := new(vppnode.VppNode)
+	changeType := resp.GetChangeType()
+	if changeType == datasync.Delete {
+		if prevValExist, err := resp.GetPrevValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+			return
+		} else if !prevValExist {
+			logging.Errorf("Delete and Previous Value Does not Exist")
+			return
+		}
+	} else {
+		if err := resp.GetValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+			return
+		}
+	}
+	ip := value.IpAddresses[0]
+	ipParts := strings.Split(ip, "/")
+	ip = ipParts[0]
+	//prefixlen, err := strconv.Atoi(ipParts[1])
+
+	client, err := remote.CreateHTTPClient("")
+	if err != nil {
+		logging.Error("create http client error: %v", err)
+		return
+	}
+
+	//Getting Ipam Info
+	b, err := getNodeInfo(client, ip, getIpamDataCmd)
+	if err != nil {
+		logging.Errorf("getnodeinfo error: %v", err)
+		return
+	}
+
+	ipam := ipv4net.IPAMData{}
+	err = json.Unmarshal(b, &ipam)
+	if err != nil {
+		logging.Errorf("unmarshal error: %v", err)
+		return
+	}
+
+	//Setting Route info
+	nlri, _ := ptypes.MarshalAny(&bgp_api.IPAddressPrefix{
+		Prefix:    ipam.PodSubnetThisNode,
+		PrefixLen: 24,
+	})
+
+	a1, _ := ptypes.MarshalAny(&bgp_api.OriginAttribute{
+		Origin: 0,
+	})
+	a2, _ := ptypes.MarshalAny(&bgp_api.NextHopAttribute{
+		NextHop: ip,
+	})
+	attrs := []*any.Any{a1, a2}
+	switch changeType {
+	case datasync.Put:
+		_, err := p.Deps.BGPServer.AddPath(context.Background(), &bgp_api.AddPathRequest{
+			Path: &bgp_api.Path{
+				Family: &bgp_api.Family{Afi: bgp_api.Family_AFI_IP, Safi: bgp_api.Family_SAFI_UNICAST},
+				Nlri:   nlri,
+				Pattrs: attrs,
+			},
+		})
+		if err != nil {
+			logging.Errorf("AddPath: %v", err)
+		}
+	case datasync.Delete:
+		err := p.Deps.BGPServer.DeletePath(context.Background(), &bgp_api.DeletePathRequest{
+			Path: &bgp_api.Path{
+				Family: &bgp_api.Family{Afi: bgp_api.Family_AFI_IP, Safi: bgp_api.Family_SAFI_UNICAST},
+				Nlri:   nlri,
+				Pattrs: attrs,
+			},
+		})
+		if err != nil {
+			logging.Errorf("AddPath: %v", err)
+		}
+	default:
+		logging.Errorf("GetChangeType: %v", changeType)
+	}
+
+}
+
+// getNodeInfo will make an http request for the given command and return an indented slice of bytes.
+//copied from vpp/plugins/netctl/cmdimpl/nodes.go
+func getNodeInfo(client *remote.HTTPClient, base string, cmd string) ([]byte, error) {
+	res, err := client.Get(base, cmd)
+	if err != nil {
+		err := fmt.Errorf("getNodeInfo: url: %s Get Error: %s", cmd, err.Error())
+		fmt.Printf("http get error: %s ", err.Error())
+		return nil, err
+	}
+	defer res.Body.Close()
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		err := fmt.Errorf("getNodeInfo: url: %s HTTP res.Status: %s", cmd, res.Status)
+		fmt.Printf("http get error: %s ", err.Error())
```

Comment:
change fmt.Printf to p.log.Info; also, put p.log.Info *before* you create err, since then you will put together where you create err and where you use it (slightly easier to read)